primefaces/src/main/java/org/primefaces/component/badge/BadgeRenderer.java

```diff
@@ -117,8 +120,22 @@ protected void encode(FacesContext context, Badge badge, BadgeModel badgeModel,
             writer.writeAttribute("style", model.getStyle(), "style");
         }
 
-        if (!valueEmpty && model.isVisible()) {
-            writer.writeText(value, "value");
+        if (model.isVisible()) {
```

Comment:
I would have delegated this to a new method, for better readability and reduced complexity (nesting).